src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -0,0 +1,122 @@
+        /// </summary>
+        public ZuneTrackCollection()
+        {
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            PART_ArtistColumn = GetTemplateChild(nameof(PART_ArtistColumn)) as TextBlock;
+            Guard.IsNotNull(PART_ArtistColumn, nameof(PART_ArtistColumn));
+
+            PART_ArtistColumn.Visibility = Visibility.Collapsed;
+        }
+
+        /// <summary>
+        /// Backing dependency property for <see cref="Collection"/>.
+        /// </summary>
+        public new ITrackCollectionViewModel Collection
+        {
+            get { return (ITrackCollectionViewModel)GetValue(CollectionProperty); }
+            set { SetValue(CollectionProperty, value); }
+        }
+
+        /// <summary>
+        /// Dependency property for <see cref="ITrackCollectionViewModel" />.
+        /// </summary>
+        public static readonly new DependencyProperty CollectionProperty =
+            DependencyProperty.Register(nameof(Collection), typeof(ITrackCollectionViewModel), typeof(ZuneTrackCollection), new PropertyMetadata(null, (s, e) =>
+            {
+                if (s is ZuneTrackCollection trackCollection)
+                {
+                    if (e.NewValue is ITrackCollectionViewModel zt)
+                    {
+                        zt.Tracks.CollectionChanged += trackCollection.Tracks_CollectionChanged;
+                    }
+
+                    if (e.OldValue is ITrackCollectionViewModel zte)
+                    {
+                        zte.Tracks.CollectionChanged -= trackCollection.Tracks_CollectionChanged;
+                    }
+                }
+            }));
+
+        /// <summary>
+        /// Backing dependency property for <see cref="AlbumArtistCollection"/>.
+        /// </summary>
+        public IAlbumCollectionViewModel? AlbumArtistCollection
+        {
+            get { return (IAlbumCollectionViewModel)GetValue(AlbumArtistCollectionProperty); }
+            set { SetValue(AlbumArtistCollectionProperty, value); }
+        }
+
+        /// <summary>
+        /// Dependency property for <ses cref="IAlbumCollectionViewModel" />.
+        /// </summary>
+        public static readonly DependencyProperty AlbumArtistCollectionProperty =
+            DependencyProperty.Register(nameof(AlbumArtistCollection), typeof(ITrackCollectionViewModel), typeof(ZuneTrackCollection), new PropertyMetadata(null, null));
+
+        private async void Tracks_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            if (PART_ArtistColumn == null)
+                return;
+
+            if (Collection is AlbumViewModel)
+            {
+                foreach (var track in Collection.Tracks)
+                {
+                    var artists = await track.GetArtistItemsAsync(track.TotalArtistItemsCount, 0).ToListAsync();
```

Comment:
Due to this line, the app will crash if the collection view is open during a file scan. You're passing this value in, but it can be 0 if the artists haven't been emitted yet. You can't use 0 as the limit value (the call needs to be skipped instead).

This also doesn't work if the artist count _changes_, you'll need to listen for that and adjust the UI appropriately.

![image](https://user-images.githubusercontent.com/9384894/172931279-4f0d7554-16f9-4f63-9aa0-0c605d876c32.png)